pytest/tests/mocknet/mirror.py

```diff
@@ -473,9 +474,24 @@ def filter_hosts(args, traffic_generator, nodes):
         logger.error(f'No hosts selected. Change filters and try again.')
         exit(1)
 
+    if args.select_partition is not None:
+        i, n = args.select_partition
+        nodes.sort(key=lambda node: node.name())
+        nodes = np.array_split(nodes, n)[i-1]
```

Comment:
if you have 5 nodes and give --select-partition 5/10 or 6/10 you'll get an empty array. I think it would make sense to add a basic sanity check somewhere if the user gives a value that doesnt make sense